VocaDbWeb/Controllers/StatsController.cs

```diff
@@ -601,7 +601,7 @@ public ActionResult SongsPerVocaloidOverTime(DateTime? cutoff, ArtistType[] voca
 			return AreaChart("Songs per voicebank over time", dataSeries);
 		}
 
-		[ResponseCache(Duration = ClientCacheDurationSec)]
+		[ResponseCache(Duration = ClientCacheDurationSec, VaryByQueryKeys = new[] { "cutoff", "vocalistTypes", "startYear" })]

```

Comment:
I thought it was silly having to repeat all parameter names for all requests, so I tried Googling. According to Google, VaryByQueryKeys should support "*" to vary by all query keys. It's very poorly documented.

So, `VaryByQueryKeys = new[] { "*" }` should work for all of these. I think it would be better for maintainability, in addition to being shorter.

Reply:
I think I also read the SO answer, but does it mean the following URLs will have different caches?

- `?cutoff=3939&vocalistTypes=Vocaloid&startYear=39`
- `?cutoff=3939&vocalistTypes=Vocaloid&startYear=39&a=1`
- `?cutoff=3939&vocalistTypes=Vocaloid&startYear=39&a=1&b=2`

`[ResponseCache]` will vary by `a` and `b`, even though they are not valid query parameters for `/Stats/GetSongsPerVoicebankTypeOverTime`?

Reply:
> I think I also read the SO answer, but does it mean the following URLs will have different caches? (I'm not sure if it's problematic.)

I think it will vary yes, because it seems to be reading the keys from query, not action. But I don't see it as problem. Normally you won't specify those parameters a and b. The only reason why you would add such dummy parameters is to intentionally skip (bust) the cache, which is also ok.

Reply:
I see, thanks for the explaination!

Reply:
I see, thanks for the explanation!

Reply:
I should still add that maybe for more public APIs I would be more careful with using the "*", but these MVC controllers are only meant to be called from the site itself, so we control both client and server.

I'm still surprised there's no way to tell ASP.NET to vary cache by only those available parameters.